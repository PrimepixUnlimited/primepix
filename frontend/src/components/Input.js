import React from 'react';
import {StyleSheet} from 'react-native';
import {Input} from 'react-native-elements';

import styles from '../constants/styles';

const InputWrapper = ({
  containerStyle,
  disabled,
  disabledInputStyle,
  inputContainerStyle,
  errorMessage,
  errorStyle,
  errorProps,
  inputComponent,
  inputStyle,
  label,
  labelStyle,
  labelProps,
  leftIconName,
  leftIconContainerStyle,
  placeholder,
  rightIconName,
  rightIconContainerStyle,
  secureTextEntry,
}) => {
  const leftIcon = {
    color: styles.greyScale.white4,
    name: leftIconName,
    size: 18,
  };
  const rightIcon = {
    color: styles.greyScale.white4,
    name: rightIconName,
    size: 18,
  };
  return (
    <Input
      containerStyle={StyleSheet.flatten([
        styles.utils.noHorizontalPadding,
        containerStyle,
      ])}
      disabled={disabled}
      disabledInputStyle={disabledInputStyle}
      inputContainerStyle={inputContainerStyle}
      errorMessage={errorMessage}
      errorStyle={errorStyle}
      errorProps={errorProps}
      inputComponent={inputComponent}
      inputStyle={inputStyle}
      label={label}
      labelStyle={labelStyle}
      labelProps={labelProps}
      leftIcon={leftIcon}
      leftIconContainerStyle={StyleSheet.flatten([
        {marginLeft: 0, marginRight: 10},
        leftIconContainerStyle,
      ])}
      placeholder={placeholder}
      placeholderTextColor={styles.greyScale.white4}
      rightIcon={rightIcon}
      rightIconContainerStyle={rightIconContainerStyle}
      secureTextEntry={secureTextEntry}
    />
  );
};

InputWrapper.defaultProps = {
  inputStyle: styles.form.input,
  labelStyle: styles.form.label,
};

export default InputWrapper;
