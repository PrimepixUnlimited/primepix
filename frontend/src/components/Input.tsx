import React from 'react';
import {StyleSheet} from 'react-native';
import {Input} from 'react-native-elements';

import styles from '../constants/styles';

const InputWrapper = ({
  containerStyle,
  disabled,
  disabledInputStyle,
  inputContainerStyle,
  errorMessage,
  errorStyle,
  errorProps,
  inputComponent,
  inputStyle,
  label,
  labelStyle,
  labelProps,
  leftIconName,
  leftIconContainerStyle,
  leftIconType,
  placeholder,
  rightIconName,
  rightIconContainerStyle,
  rightIconType,
  secureTextEntry,
  ...props
}) => {
  const leftIcon = {
    color: styles.greyScale.white4,
    name: leftIconName,
    size: 18,
    type: leftIconType,
  };
  const rightIcon = {
    color: styles.greyScale.white4,
    name: rightIconName,
    size: 18,
    type: rightIconType,
  };
  return (
    <Input
      {...props}
      containerStyle={StyleSheet.flatten([
        styles.utils.noHorizontalPadding,
        containerStyle,
      ])}
      disabled={disabled}
      disabledInputStyle={disabledInputStyle}
      inputContainerStyle={StyleSheet.flatten([
        inputContainerStyle,
        {borderColor: styles.greyScale.black2},
      ])}
      errorMessage={errorMessage}
      errorStyle={errorStyle}
      errorProps={errorProps}
      inputComponent={inputComponent}
      inputStyle={StyleSheet.flatten([inputStyle, {borderColor: '#ccc'}])}
      label={label}
      labelStyle={labelStyle}
      labelProps={labelProps}
      leftIcon={leftIcon}
      leftIconContainerStyle={StyleSheet.flatten([
        {marginLeft: 0, marginRight: 10},
        leftIconContainerStyle,
      ])}
      placeholder={placeholder}
      placeholderTextColor={styles.greyScale.white4}
      rightIcon={rightIcon}
      rightIconContainerStyle={rightIconContainerStyle}
      secureTextEntry={secureTextEntry}
    />
  );
};

InputWrapper.defaultProps = {
  inputStyle: styles.form.input,
  labelStyle: styles.form.label,
};

export default InputWrapper;
