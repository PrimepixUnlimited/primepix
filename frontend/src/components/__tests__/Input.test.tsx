// components/__tests__/Hello.tsx
import "react-native";
import * as React from "react";
import * as renderer from "react-test-renderer";

import Input from "../Input";

describe("Input snapshot", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <Input
          label="Expiry month"
          leftIconName="calendar"
          leftIconType="material-community"
          onChangeText={() => {}}
          placeholder="MM"
          value={"test"}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
