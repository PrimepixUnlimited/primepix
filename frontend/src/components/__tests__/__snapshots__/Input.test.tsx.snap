// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 0,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Expiry month
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#202124",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 0,
          "marginRight": 10,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onChangeText={[Function]}
      placeholder="MM"
      placeholderTextColor="#3b423b"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#ccc",
          "color": "#fff",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
      value="test"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
